dashboard/app/models/level.rb

```diff
@@ -349,6 +351,27 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  def log_changes
+    log = JSON.parse(audit_log || "[]")
+
+    # serialize 'properties' changes
+    changed = self.changed
+    if changed.include?('properties')
+      changed.delete('properties')
+      changed_attributes['properties'].each do |key, value|
+        changed.push(key) if properties[key] != value
+      end
+    end
```

Comment:
Because most of the actual information about the level (the map data, instructions, block xml, etc) is serialized into a JSON blob stored in `properties`, if we _didn't_ do this then the vast majority of log entries we see would just show that `properties` changed; with this, we can peek into the JSON blob to see which entries changed.